config/initializers/controllers.rb

```diff
@@ -69,4 +72,21 @@ def disable_fine_print
   def set_locale
     I18n.locale = http_accept_language.compatible_language_from(I18n.available_locales)
   end
+
+  def set_device_id
+    cookies.delete(:oxdid) if cookies[:oxdid] && !(cookies[:oxdid] =~ UUID_REGEX)
```

Comment:
non-trivial if expressions are often better written as multiline.  this is close, but, imo, it'd be a bit easier to grok if multi line. 

https://github.com/airbnb/ruby#only-simple-if-unless

Reply:
would be nice if this piece were encapsulated in a class w/ a SRP on its own (and unit tested in isolation from the controller)

Reply:
and then a simpler golden path test for the controller

Reply:
made the `if` better via a new helper method